servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java

```diff
@@ -351,10 +359,16 @@ public BlockMeta moveBlockMeta(BlockMeta blockMeta, TempBlockMeta tempBlockMeta)
       throws NotFoundException, OutOfSpaceException, AlreadyExistsException {
     StorageDir srcDir = blockMeta.getParentDir();
     StorageDir dstDir = tempBlockMeta.getParentDir();
+
+    dstDir.removeTempBlockMeta(tempBlockMeta);
+
+    if (srcDir.equals(dstDir)) {
+      return blockMeta;
+    }
+
```

Comment:
On the whole, there are only two case:
case1: top-level called by `BlockServiceHandler#promoteBlock()` for promoting block to top tier, should never reach;
case2: called by `TieredBlockStore#freeSpaceInternal`, should never reach here too;
While from the view this single API I think it's reasonable to do this check.